rpcs3/rpcs3qt/main_window.cpp

```diff
@@ -2282,6 +2225,97 @@ void main_window::ShowTitleBars(bool show) const
 	m_log_frame->SetTitleBarVisible(show);
 }
 
+void main_window::ShowOptionalGamePreparations(const QString& title, const QString& message, std::map<std::string, QString> bootable_paths)
+{
+	QDialog* dlg = new QDialog(this);
+	dlg->setObjectName("game_prepare_window");
+	dlg->setWindowTitle(title);
+
+	QVBoxLayout* vlayout = new QVBoxLayout(dlg);
+
+	QCheckBox* desk_check = new QCheckBox(tr("Add desktop shortcut(s)"));
+#ifdef _WIN32
+	QCheckBox* quick_check = new QCheckBox(tr("Add Start menu shortcut(s)"));
+#elif defined(__APPLE__)
+	QCheckBox* quick_check = new QCheckBox(tr("Add dock shortcut(s)"));
+#else
+	QCheckBox* quick_check = new QCheckBox(tr("Add launcher shortcut(s)"));
+#endif
+	QCheckBox* precompile_check = new QCheckBox(tr("Precompile caches"));
+	QLabel* label = new QLabel(tr("%1\nWould you like to install shortcuts to the installed software and precompile caches? (%2 new software detected)\n\n").arg(message).arg(bootable_paths.size()), dlg);
+
+	vlayout->addWidget(label);
+	vlayout->addStretch(10);
+	vlayout->addWidget(desk_check);
+	vlayout->addStretch(3);
+	vlayout->addWidget(quick_check);
+	vlayout->addStretch(3);
+	vlayout->addWidget(precompile_check);
+	vlayout->addStretch(3);
+
+	precompile_check->setToolTip(tr("Spend time building data needed for game boot now instead of at launch."));
+
+	QDialogButtonBox* btn_box = new QDialogButtonBox(QDialogButtonBox::Ok);
+
+	vlayout->addWidget(btn_box);
+	dlg->setLayout(vlayout);
+
+	connect(btn_box, &QDialogButtonBox::accepted, this, [this, dlg, desk_check, quick_check, precompile_check, bootable_paths = std::move(bootable_paths)]()
```

Comment:
This dialog should be its own class that connects to this main window function instead of adding more and more and more nested dialog code to big classes like this.